packages/react-storage/src/components/StorageBrowser/views/LocationDetailView/useLocationDetailView.ts

```diff
@@ -117,12 +118,16 @@ export function useLocationDetailView(
     return result.slice(start, end);
   }, [range, result]);
 
+  const isFinalPage =
+    !hasNextToken && isLastPage(currentPage, resultCount, pageSize);
+  const hasNoResults = pageItems.length === 0;
+
   return {
     page: currentPage,
     pageItems,
-    hasNextPage: hasNextToken,
-    isPaginateNextDisabled: !hasNextToken || isLoading || hasError,
-    isPaginatePreviousDisabled: currentPage <= 1 || isLoading || hasError,
+    isPaginateNextDisabled:
```

Comment:
These will be moved to the pagination control in [upcoming PRs](https://github.com/aws-amplify/amplify-ui/pull/5955) but fixing it here for now